packages/netsuite-adapter/src/filters/element_references.ts

```diff
@@ -108,6 +116,51 @@ const generateServiceIdToElemID = async (
   .map(elem => getElementServiceIdRecords(elem))
   .reduce<ServiceIdRecords>((acc, records) => Object.assign(acc, records), {})
 
+const isFilePath = (path: string): boolean =>
+  osPath.extname(osPath.basename(path)) !== ''
+
+const resolveRelativePath = (absolutePath: string | undefined, relativePath: string): string =>
+  (isDefined(absolutePath) ? osPath.resolve(osPath.dirname(absolutePath), relativePath) : '')
+
+const updateDependenciesToAdd = (
+  foundReferences: string[],
+  fetchedElementsServiceIdToElemID: ServiceIdRecords,
+  elementsSourceServiceIdToElemID: ServiceIdRecords,
+  element: InstanceElement,
+): ElemID[] =>
+  foundReferences
+    .map(ref => (isFilePath(ref) ? resolveRelativePath(element.value?.path, ref) : ref))
+    .map(ref => {
+      const serviceIdRecord = fetchedElementsServiceIdToElemID[ref] ?? elementsSourceServiceIdToElemID[ref]
+      if (isDefined(serviceIdRecord) && typeof serviceIdRecord !== 'function') {
```

Comment:
how will it be a function?

Reply:
In cases where the reference is a string that matches a function name of Object.prototype the maps return a function, encountered it when the string 'toString' appeared in one of the files.

Reply:
Wow. I didn't know it works that way haha. Maybe this is a nicer way-
```suggestion
      if (_.isPlainObject(serviceIdRecord)) {
```